--- a/include/Makefile.skel
+++ b/include/Makefile.skel
@@ -919,7 +919,7 @@ 
 #CRAYCNL#TAU_FORTRANLIBS = #ENDIF#
 #AOCC#TAU_FORTRANLIBS	       = -lflangmain -lflang -lflangrti -lompstub -lm -lrt -lpgmath -lomp #ENDIF#
 #NVIDIA#TAU_FORTRANLIBS	       = $(EXTRADIR)/lib/f90main.o -L$(EXTRADIR)/lib -lnvf -Wl,-rpath,$(EXTRADIR)/lib  #ENDIF#
-#INTEL81FIX#TAU_FORTRANLIBS       = -L$(TAUGCCLIBDIR) $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) -lcxa -lunwind -L$(EXTRADIR)/lib -lifcore $(EXTRADIR)/lib/for_main.o #ENDIF#
+#INTEL81FIX#TAU_FORTRANLIBS       = -L$(TAUGCCLIBDIR) $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) -lunwind -L$(EXTRADIR)/lib -lifcore $(EXTRADIR)/lib/for_main.o #ENDIF# 
 #INTEL10FIX#TAU_FORTRANLIBS       = -L$(TAUGCCLIBDIR) $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) -L$(IFORTLIBDIR) -lifcore $(IFORTLIBDIR)/for_main.o #ENDIF#
 #INTEL_ICPX#TAU_FORTRANLIBS       = -L$(TAUGCCLIBDIR) $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) -L$(IFORTLIBDIR) -lifcore $(IFORTLIBDIR)/for_main.o #ENDIF#
 #INTELCXXLIBICC#TAU_FORTRANLIBS       = -lcprts -L$(EXTRADIR)/lib -lifcore $(EXTRADIR)/lib/for_main.o #ENDIF#
@@ -1042,7 +1042,7 @@ 
 #USE_INTELCXX#TAU_CXXLIBS_INTEL        = -lcprts -lPEPCF90 #ENDIF#
 #USE_INTELCXX#TAU_CXXLIBS_INTEL        = -lcprts #ENDIF#
 #INTELIFORT#TAU_CXXLIBS_INTEL        = -lcprts #ENDIF#
-#INTEL81FIX#TAU_CXXLIBS_INTEL        = -L$(TAUGCCLIBDIR) $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) -lcxa -lunwind#ENDIF#
+#INTEL81FIX#TAU_CXXLIBS_INTEL        = -L$(TAUGCCLIBDIR) $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) -lunwind#ENDIF#
 #INTEL10FIX#TAU_CXXLIBS_INTEL        = -L$(TAUGCCLIBDIR) $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) #ENDIF#
 #INTEL_ICPX#TAU_CXXLIBS_INTEL        = $(TAUGCCLIBOPTS) -lstdc++ $(TAU_GCCLIB) #ENDIF#
 #INTELCXXLIBICC#TAU_CXXLIBS_INTEL        = -lcprts #ENDIF#
