--- a/src/Profile/Makefile.wrapper
+++ b/src/Profile/Makefile.wrapper
@@ -63,7 +63,7 @@ TauFMpi.o : TauFMpi.c
 	$(CC) $(TAU_MPI_INCLUDE)  $(CFLAGS) -c TauFMpi.c
 
 TauFInit.o : TauFInit.f
-	$(TAU_F90) $(TAU_INTERNAL_FLAGS) $(TAU_MPI_INC) $(FFLAGS) $(PROFILEOPT10) -c TauFInit.f
+	$(TAU_F90) $(TAU_INTERNAL_FLAGS) $(TAU_MPI_INC) $(FFLAGS) $(PROFILEOPT10) $(USER_OPT) -c TauFInit.f
 
 TauMpiExtensions.o : 	TauMpiExtensions.c
 	$(CC) $(TAU_MPI_INCLUDE)  $(CFLAGS) -c TauMpiExtensions.c
