Allow running with Python 3.10+
diff --git a/bin/parseDirectiveArgs.py b/bin/parseDirectiveArgs.py
index 0fbf4a6..11a6e18 100755
--- a/bin/parseDirectiveArgs.py
+++ b/bin/parseDirectiveArgs.py
@@ -12,7 +12,7 @@ def flatten(l):
     """
     if l:
         for el in l:
-            if isinstance(el, collections.Iterable) \
+            if isinstance(el, collections.abc.Iterable) \
                and not isinstance(el, (str, bytes)):
                 # The following is incompatible with python 3.
                 # if isinstance(el, collections.Iterable) \
