Index: extern/remap/src/earcut.hpp
===================================================================
--- extern/remap/src/earcut.hpp	(revision 2252)
+++ extern/remap/src/earcut.hpp	(working copy)
@@ -6,6 +6,9 @@
 #include <memory>
 #include <vector>
 #include <limits>
+// Using stdint.h instead of cstdint ensures that the symbols
+// are in the global namespace ... which is how uint32_t is used
+#include <stdint.h>
 //#include <tuple>
 //#include <cstdint> 
 //#include <cstddef>
Index: extern/remap/src/meshutil.cpp
===================================================================
--- extern/remap/src/meshutil.cpp	(revision 2252)
+++ extern/remap/src/meshutil.cpp	(working copy)
@@ -4,6 +4,7 @@
 #include "intersection_ym.hpp"
 #include "earcut.hpp"
 #include <vector>
+#include <array>
 
 namespace sphereRemap {
 
