Index: extern/remap/src/earcut.hpp
===================================================================
--- a/extern/remap/src/earcut.hpp       (revision 2252)
+++ b/extern/remap/src/earcut.hpp       (working copy)
@@ -6,6 +6,9 @@
#include <memory>
#include <vector>
#include <limits>
+// Using cstdint ensures that the symbols are in the
+// global namespace and in std ... which is how uint32_t is used
+#include <cstdint>
//#include <tuple>
//#include <cstdint> 
//#include <cstddef>
Index: extern/remap/src/meshutil.cpp
===================================================================
--- a/extern/remap/src/meshutil.cpp     (revision 2252)
+++ b/extern/remap/src/meshutil.cpp     (working copy)
@@ -4,6 +4,7 @@
#include "intersection_ym.hpp"
#include "earcut.hpp"
#include <vector>
+#include <array>

namespace sphereRemap {

